src/Application.php

```diff
@@ -619,6 +639,20 @@ public function getFinalHandler(ResponseInterface $response = null)
     }
 
     /**
+     * Return the default delegate to use during `run()` when the stack is exhausted.
+     *
+     * Provided as a forwards compatibility measure with version 2, and proxies
+     * to getFinalHandler(); please note that it does not accept any arguments,
+     * nor pass them on to that method.
+     *
+     * @return callable|null
+     */
+    public function getDefaultDelegate()
```

Comment:
Sure: you could be nesting an application within another, and want to re-use the default delegate from the parent application. As such, you might build the factory like this:

```php
use Zend\Expressive\AppFactory;
use Zend\Expressive\Application;
use Zend\Expressive\Router\FastRouteRouter;

function ($container)
{
    $app = $container->get(Application::class);
    $nested = new Application(
        new FastRouteRouter(),
        $container,
        $app->getDefaultDelegate()
    );
    
    /* pipe and/or route some middleware */
    $nested->pipe(/* ... */);
    $nested->get('/subpath', /* ... */);
    
    return $nested;
}
```

In normal circumstances, if this were dispatched, if a call to `$next()`/`$delegate->process()` occurs but the pipeline is exhausted, this would return handling to the parent, which might lead to execution later down the stack. By passing the default delegate in, we ensure that this nested application then becomes terminal; once it is dispatched, it _must_ return a response, ensuring no deeper layers are executed.